ports/atmel-samd/Makefile

```diff
@@ -23,8 +23,15 @@
 # THE SOFTWARE.
 
 # Select the board to build for.
+define boardlist
+
+$(foreach dir,$(filter-out %.*,$(wildcard boards/*)),$(notdir $(dir)))
+
+
+endef
+
 ifeq ($(BOARD),)
-  $(error You must provide a BOARD parameter)
+  $(error You must provide a BOARD parameter: $(boardlist)Rerun with $(MAKE) BOARD=<board>)
```

Comment:
I started out just wanting to request that a comment be added that the blank lines within `boardlist` are important and cannot just be removed.

But then I got creative. I don't know if `column` is standard on macs, but I came up with this:
```make
define show_board_error
boardlist = 
$(info Valid boards:)
$(shell printf '%s\n' $(patsubst boards/%/mpconfigboard.mk,%,$(wildcard boards/*/mpconfigboard.mk)) | column -c 80 1>&2)
$(error Rerun with $(MAKE) BOARD=<board>)
endef

ifeq ($(BOARD),)
    $(info No BOARD specified)
    $(call show_board_error)
else
  ifeq ($(wildcard boards/$(BOARD)/.),)
    $(info Invalid BOARD specified)
    $(call show_board_error)
  endif
endif
```

Reply:
I am absolutely good with adding this for the rest of the Makefiles.
PS - I was trying to use column before sending the last commit of the night, but couldn't quite get make to preserve the newlines.

Yes, Column is available.. But we could do 
```COLUMN=$(shell which column || echo cat)```